docsrc/interactivity.nim

```diff
@@ -2,57 +2,125 @@ import nimib
 
 nbInit
 
+template nimibCode*(body: untyped) =
+  newNbCodeBlock("nimibCode", body):
+    discard
+  body
+
+nb.partials["nimibCode"] = nb.partials["nbCode"]
+nb.renderPlans["nimibCode"] = nb.renderPlans["nbCode"]
+
 nbText: hlMd"""
-# Creating interactive components in Nimib
+# Creating interactive components in nimib
 
 Nimib can easily be used to create static content with `nbText` and `nbCode`, but did you know that you can create interactive
 content as well? And that you can do it all in Nim even! This can be achieved using either the `nbJsFromCode`-API or `nbKaraxCode`.
 They work by compiling Nim code into javascript and adding it to the resulting HTML file.
 This means that arbitrary Javascript can be written but also that Karax, which compiles to javascript, also can be used.
+The available template are:
```

Comment:
I would actually start with the "by default we compile all into a single js file". 
A start could be:

```md
In the same way that code from nbCode blocks are all compiled into a single file,
all code to be compiled in javascript will be put in a single file. 
This has the advantage that a single compilation is performed
and code from a previous block can be used in subsequent blocks.
The api looks like this:

- `nbJsFromCode`: nim code will be appended to the file and compiled during `nbSave`.
- `nbJsFromCodeInBlock`: same as `nbJsFromCode` but the code is put inside a `block`.
- `nbJsFromCodeGlobal`: the code here will be put at the top of the file inside the global code.

If you wish to compile to a separate file you can do that.
Indeed this is what is done for a special block that allows you to use karax removing all boilerplate:

- `nbJsFromCodeOwnFile`: compile to js as its own file.
- `nbKaraxCode`: Sugar on top of `nbJsFromCodeOwnFile` for writing Karax components.
```

Reply:
Great idea, thanks :D Rephrased it in a few places